src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2532,160 +2653,246 @@ var projectUniqueNamesForBuildIntToUpdate
-                    updatedPackageSpec,
-                    firstAction.PackageIdentity,
-                    successfulFrameworks);
-
-                restoreResult = await DependencyGraphRestoreUtility.PreviewRestoreAsync(
-                    SolutionManager,
-                    buildIntegratedProject,
-                    updatedPackageSpec,
-                    dependencyGraphContext,
-                    providerCache,
-                    cacheModifier,
-                    sources,
-                    nuGetProjectContext.OperationId,
-                    logger,
-                    token);
-            }
-
-            // If HideWarningsAndErrors is true then restore will not display the warnings and errors.
-            // Further, replay errors and warnings only if restore failed because the assets file will not be committed.
-            // If there were only warnings then those are written to assets file and committed. The design time build will replay them.
-            if (updatedPackageSpec.RestoreSettings.HideWarningsAndErrors &&
-                !restoreResult.Result.Success)
+            foreach (var buildIntegratedProject in buildIntegratedProjects)
             {
-                await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(restoreResult.Result.LockFile?.LogMessages, logger);
-            }
+                var nuGetProjectActions = nugetProjectActionsLookup[buildIntegratedProject.MSBuildProjectPath];
+                var nuGetProjectActionsList = nuGetProjectActions.ToList();
+                var updatedPackageSpec = updatedNugetPackageSpecLookup[buildIntegratedProject.MSBuildProjectPath];
+                var originalPackageSpec = originalNugetPackageSpecLookup[buildIntegratedProject.MSBuildProjectPath];
+                var originalLockFile = lockFileLookup[buildIntegratedProject.MSBuildProjectPath];
+                var sources = nuGetProjectSourceLookup[buildIntegratedProject.MSBuildProjectPath];
 
-            // Build the installation context
-            var originalFrameworks = updatedPackageSpec
-                .TargetFrameworks
-                .ToDictionary(x => x.FrameworkName, x => x.TargetAlias);
+                var allFrameworks = updatedPackageSpec
+                    .TargetFrameworks
+                    .Select(t => t.FrameworkName)
+                    .Distinct()
+                    .ToList();
 
-            var installationContext = new BuildIntegratedInstallationContext(
-                successfulFrameworks,
-                unsuccessfulFrameworks,
-                originalFrameworks);
+                var restoreResult = restoreResults.Single(r =>
+                    string.Equals(
+                        r.SummaryRequest.Request.Project.RestoreMetadata.ProjectPath,
+                        buildIntegratedProject.MSBuildProjectPath,
+                        StringComparison.OrdinalIgnoreCase));
+
+                var unsuccessfulFrameworks = restoreResult
```

Comment:
In the case of a restoreResult.Result.Success == true why is needed to calculate the unsuccesulFrameworks?
If the restoreResult.Result.Success == true, should not be unsuccesulFrameworks be empty and successfulFrameworks  == allFrameworks?

Maybe some linq queries execution could be eliminated?

Reply:
Even though we eliminate here still we need both them https://github.com/NuGet/NuGet.Client/blob/24c713cd927fe4920ec5ecc6fe98033f84eab727/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2855-L2857 as parameter soon after. So we can't eliminate it.

Reply:
But we can eliminate the linq queries execution. unsuccessfulFrameworks would be empty list and the successful will be the entire allFrameworks list.  The previous code(before your changes) was calculating the queries as well, so I will let it up to you.

Reply:
Just in case I want to keep it this way, I suspect business logic might change. I want gradual changes, not too much once.